### PR TITLE
Update lifetimes.md

### PR DESCRIPTION
**Pull request by kineticz**
Based on a discussion on IRC, this should be the correct expansion.

### CODE REVIEW

**Comment by Manishearth** [comment]
Could you link to the discussion? `'a` is correct here.

**Comment by kindlychung** [comment]
I have pinged @bluss .

File: src/doc/book/lifetimes.md [hunk 1]
```diff
@@ -355,5 +355,5 @@ fn args<T: ToCStr>(&mut self, args: &[T]) -> &mut Command; // elided
 fn args<'a, 'b, T: ToCStr>(&'a mut self, args: &'b [T]) -> &'a mut Command; // expanded
 
 fn new(buf: &mut [u8]) -> BufWriter; // elided
```
Comment (mbrubeck) [inline src/doc/book/lifetimes.md]: This line should be:

```rust
  fn new(buf: &mut [u8]) -> BufWriter<&mut [u8]>; // elided 
```

**Comment by bluss** [comment]
The elided/expanded lines were consistent but they don't match the BufWriter in libstd. Best update both lines so that they do. Either that, or pick a simpler example. For example [str::lines](http://doc.rust-lang.org/nightly/std/primitive.str.html#method.lines) (anything goes really).

**Comment by Manishearth** [comment]
Closing, feel free to reopen with commends addressed. Thanks!

**Pull request closed without merging**